extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/LiquibaseFactory.java

```diff
@@ -31,9 +34,23 @@ public Liquibase createLiquibase() {
         try (ClassLoaderResourceAccessor resourceAccessor = new ClassLoaderResourceAccessor(
                 Thread.currentThread().getContextClassLoader())) {
 
-            Database database = DatabaseFactory.getInstance()
-                    .findCorrectDatabaseImplementation(new JdbcConnection(dataSource.getConnection()));
-            ;
+            Database database;
+
+            if (config.username.isPresent() && config.password.isPresent()) {
```

Comment:
The Javadoc for `username` says: 
- `If no specific JDBC URL is configured, falls back to the datasource username if not specified.`
- 
And for `password`: 
- `If no specific JDBC URL is configured, falls back to the datasource password if not specified.`
 
This seems confusing as they are both ignored if one is not present.